Classes/Common/DocumentList.php

```diff
@@ -316,6 +252,122 @@ protected function getRecord($element)
         return $record;
     }
 
+    /**
+     * It gets SOLR result
+     *
+     * @access private
+     *
+     * @param array $record: for searched document
+     *
+     * @return Result
+     */
+    private function getSolrResult($record) {
+        $fields = Solr::getFields();
+
+        $query = $this->solr->service->createSelect();
+        // Restrict the fields to the required ones
+        $query->setFields($fields['uid'] .',' . $fields['id'] .',' . $fields['toplevel'] .',' . $fields['thumbnail'] .',' . $fields['page']);
+        foreach ($this->solrConfig as $solr_name) {
+            $query->addField($solr_name);
+        }
+        // Set additional query parameters.
+        // Set reasonable limit for safety reasons.
+        // We don't expect to get more than 10.000 hits per UID.
+        $query->setStart(0)->setRows(1000);
+        // Take over existing filter queries.
+        $filterQueries = isset($this->metadata['options']['params']['filterquery']) ? $this->metadata['options']['params']['filterquery'] : [];
+        // Extend filter query to get all documents with the same UID.
+        foreach ($filterQueries as $key => $value) {
+            if (isset($value['query'])) {
+                $filterQuery[$key] = $value['query'] . ' OR ' . $fields['toplevel'] . ':true';
+                $query->addFilterQuery($filterQuery);
+            }
+        }
+        // Add filter query to get all documents with the required uid.
+        $filterQuery['key'] = $fields['uid'] . ':' . Solr::escapeQuery($record['uid']);
+        $query->addFilterQuery($filterQuery);
+        // Add sorting.
+        $query->addSort('score', $this->metadata['options']['params']['sort']['score']);
+        // Set query.
+        $query->setQuery($this->metadata['options']['select'] . ' OR ' . $fields['toplevel'] . ':true');
+
+        // If it is a fulltext search, enable highlighting.
+        if ($this->metadata['fulltextSearch']) {
+            $query->setQuery(
+                $fields['fulltext'] . ':' . Solr::escapeQuery($this->metadata['searchString']));
+                // TODO:toplevel documents throw error in SOLR PLugin:
```

Comment:
I'm not sure, if this is the reason for the empty toplevel element in listview. Could you please have a look at this:

![Listview](https://user-images.githubusercontent.com/3354500/129206463-6962b640-2c4e-4c8b-8732-225a27ff1cec.png)

That's how it should look like:

![Listview (1)](https://user-images.githubusercontent.com/3354500/129206496-c99c40de-b6e3-414c-b0b9-e272ee6e4043.png)

Below the toplevel element, the pages are shown. In the current state, the text snippets are shown again. That's perfect.

Reply:
Yes there is a problem with top element. It does't return any result if fulltext is empty as it throws error in SOLR Plugin.